helix-term/src/ui/statusline.rs

```diff
@@ -179,16 +189,30 @@ where
                 "   "
             }
         ),
-        if visible && config.color_modes {
-            match context.editor.mode() {
-                Mode::Insert => Some(context.editor.theme.get("ui.statusline.insert")),
-                Mode::Select => Some(context.editor.theme.get("ui.statusline.select")),
-                Mode::Normal => Some(context.editor.theme.get("ui.statusline.normal")),
-            }
-        } else {
-            None
-        },
+        mode_style,
     );
+
+    match config.statusline.mode_separator {
+        ModeSeparator::Flat => {}
+        _ => {
+            // invert the mode style background and foreground
+            let mode_separator_style = mode_style.map(|s| {
+                let mut c = s.clone();
+                c.fg = s.bg;
+                c.bg = s.fg;
```

Comment:
Instead of using the mode's `fg` this should be the `bg` of `ui.statusline` to render correctly. Currently it displays like this:

![image](https://user-images.githubusercontent.com/3957610/206424496-8e7847c4-3340-49df-8217-65e0d9d9af4b.png)

Reply:
Fixed and tested with other themes, commit: cf679f0.